x-pack/test/api_integration/apis/search/session.ts

```diff
@@ -152,20 +153,23 @@ export default function ({ getService }: FtrProviderContext) {
 
         const { id: id2 } = searchRes2.body;
 
-        const resp = await supertest
-          .get(`/internal/session/${sessionId}`)
-          .set('kbn-xsrf', 'foo')
-          .expect(200);
-
-        const { name, touched, created, persisted, idMapping } = resp.body.attributes;
-        expect(persisted).to.be(true);
-        expect(name).to.be('My Session');
-        expect(touched).not.to.be(undefined);
-        expect(created).not.to.be(undefined);
-
-        const idMappings = Object.values(idMapping).map((value: any) => value.id);
-        expect(idMappings).to.contain(id1);
-        expect(idMappings).to.contain(id2);
+        await retry.waitForWithTimeout('searches persisted into session', 5000, async () => {
```

Comment:
We anyway had a subtle race condition here before this pr, because we never awaited `trackId` inside session service. No time to track increase because of batching, so this test was failing more ofter